lib/process.js

```diff
@@ -78,5 +98,20 @@ module.exports = function (src, path) {
     }
   }
 
+  if (Array.isArray(parts.styles) && parts.styles.length > 0) {
+    const styleStr = parts.styles.map(ast => {
+      const moduleName = ast.module || '$style'
+      const styleObj = processStyle(ast)
+
+      return '\n  this[\'' + moduleName + '\'] = ' + JSON.stringify(styleObj)
+    })
+
+    output += '\n;(function() {' +
+      '\nvar render = __vue__options__.render' +
+      '\n__vue__options__.render = function(h) {' + styleStr +
```

Comment:
Instead of adding it to the render function, I think we should create a beforeCreate() callback that adds the property during intialization of the component.

That's essentially what vue-loader does, it would look something like this:

```javascript
output += '\n;(function() {' +
      '\nvar existing = ___vue__options__.beforeCreate' +
      '\nvar styleFn =  function () { ' + styleStr + ' })' +
      '\n__vue__options__.beforeCreate = existing ? [].concat(existing, styleFn) : [styleFn]' +
      '})()'
```

This would ensure that the style is also available before render, e.g. to be accessed from within some method or a `created()` hook.